packages/core/micro/src/sheetlet.ts

```diff
@@ -615,8 +618,12 @@ function wrapIMatrix(matrix: IMatrix): IMatrixProducer<Value> {
                 : raw;
         },
         openMatrix() { return producer },
-        removeMatrixConsumer() { },
+        closeMatrix() { },
+        matrixProducer: undefined as any as IMatrixProducer<Value>,
     }
+
+    producer.matrixProducer = producer;
```

Comment:
The circular reference is a bit of a pain here.

I have a change coming that adds Value/Vector/MatrixProducer base classes to track consumers, etc...  so hopefully most people won't be conjuring producers from object literals.

Reply:
Nevermind.  I had forgotten you can declare property accessors on object literals.